src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -52,38 +52,92 @@ uint8_t getTimerIndex(TIM_TypeDef *timer)
 void pwmWriteDshotInt(uint8_t index, uint16_t value)
 {
     motorDmaOutput_t *const motor = &dmaMotors[index];
-
+#ifdef USE_DSHOT_DMAR
+    if (!motor->timerHardware || !motor->timerHardware->dmaTimUPRef) {
+        return;
+    }
+#else
     if (!motor->timerHardware || !motor->timerHardware->dmaRef) {
         return;
     }
+#endif
 
     uint16_t packet = prepareDshotPacket(motor, value);
 
     uint8_t bufferSize = loadDmaBuffer(motor, packet);
```

Comment:
It may be better to `loadDmaBuffer` into correct place in buffer directly. But I'm not that sure ..

`loadDmaBuffer` may take pointer + stride instead of motor. But it will be slightly complicated by different target types.